### ISSUE TITLE
File is not created from Template

### ISSUE THREAD

**Issue by TerrySlack** [jantimon/html-webpack-plugin#1213]
## Expected behaviour
Template is used to create a new file, with scripts injected accordingly

## Current behaviour
Template is found, build completes successfully.  However, the new file is not created.  I'm trying to use a _LayoutTemlate.cshtml file as my template and it will not create a new file named _Layout.cshtml

Tell us what the html-webpack-plugin does instead.
Nothing.  No new file is created.

## Environment
Windows 10
Node -v 12.2.0
npm -v 6.9.0
Webpack -v 4.31.0
"html-webpack-plugin -v 3.2.0

### Config

Copy the minimal `webpack.config.js` to produce this issue:

```js
const 
    path = require('path'),
    HtmlWebpackPlugin = require('html-webpack-plugin');
module.exports = {
    entry: './src/index.js',
    module: {
        rules: [
          {
            test: /\.(js|jsx)$/,
            exclude: /node_modules/,
            use: ['babel-loader']
          }
        ]
      },
      resolve: {
        extensions: ['*', '.js', '.jsx', ".cshtml"]
      },
    output: {
      path: __dirname + '/dist',
      publicPath: '/',
      filename: 'bundle.js'
    },      
    plugins: [
        new HtmlWebpackPlugin({
            inject: "body",
            template: "./src/views/shared/_LayoutTemplate.cshtml",
            filename: path.resolve("./src/views/shared/_Layout.cshtml")
        })
    ]
  };
```

Copy your template file if it is part of this issue:

```html
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>@ViewBag.Title - My ASP.NET Application</title>
    <link rel="stylesheet" type="text/css" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" />
</head>
<body>
    <div class="navbar navbar-inverse navbar-fixed-top">
        <div class="container">
            <div class="navbar-header">
                <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                    <span class="icon-bar"></span>
                </button>
                @Html.ActionLink("Application name", "Index", "Home", new { area = "" }, new { @class = "navbar-brand" })
            </div>
            <div class="navbar-collapse">
                <ul class="nav navbar-nav">
                    <li>@Html.ActionLink("Home", "Index", "Home")</li>
                    <li>@Html.ActionLink("About", "About", "Home")</li>
                    <li>@Html.ActionLink("Contact", "Contact", "Home")</li>
                </ul>
            </div>
        </div>
    </div>
    <div class="container body-content">
        @RenderBody()
        <hr />
        Honkey Tonk Hero - Phil Daigle
        <footer>
            <p>&copy; @DateTime.Now.Year - My ASP.NET Application</p>
        </footer>
    </div>

    <!--Loading external scripts to keep the webpack build down-->
    <script defer src="https://unpkg.com/react@16/umd/react.production.min.js"></script>
    <script defer src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js"></script>
    <script defer src="https://cdnjs.cloudflare.com/ajax/libs/prop-types/15.7.2/prop-types.min.js"></script>
    <script defer src="http://code.jquery.com/jquery-3.4.1.slim.min.js"></script>
    <script defer src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.15.0/umd/popper.min.js"></script>
    <script defer src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.bundle.min.js"></script>
</body>
</html>

```

## Relevant Links

Repo: https://github.com/TerrySlack/Html-Webpack-Template-Issue.git

**Comment by Yangfan2016** [comment]
mark

**Comment by riatzukiza** [comment]
I have this problem too.

**Comment by TerrySlack** [comment]
As nobody has responded to the issue, here's the soluton I found [app-root-path](https://www.npmjs.com/package/app-root-path).  I found the root path and then determined where my files needed to be accessed for templating and the result sent to.  As I have decoupled my front end project from my .net project, I needed to be able to go above the root deteced by path.resolve.  If you need an example on how to use this, post here and I'll put something up on the repo I created above.

**Comment by gavinbarron** [comment]
@TerrySlack, if you could post an example I'd really appreciate it. I've been struggling with template generation

**Comment by TerrySlack** [comment]
@gavinbarron I'll post it at some point today.  Fixing some bugs on my own webpack files and will release them asap, with a working example and explanation, in a Github repo.

**Comment by TerrySlack** [comment]
@gavinbarron I'm bogged down right now.  However, take a look at the pacakge [app-root-path](https://www.npmjs.com/package/app-root-path).  I used that, to basically find out where I want the template to be sent to.  In  my case, I use index.ejs (it contains html), within the src folder.  I then give  the HtmlWebpackPlugin the follwoing:

`const 
   appRoot = require('app-root-path'),
   root = appRoot.path.toLocaleLowerCase()`;

Then in the HtmlWebpackPlugin configuration.  I use the root.  solutionName and projectName are part of a Visual Studio solution.  So I tell the plugin to grab the _LayoutTemplate.cshtml.  Do whatever I want with it, and then to put the final version, called _Layout.cshtml into the folder, I've indicated in filename.

In my case, I have decoupled the Front End from the Visual Studio solution completely.  The FrontEnd folder sits side by side with the VS folder.  So I needed to go outside of what path would give me in Node.  Hope this helps.  Let me know if you have any issues.

 `new HtmlWebpackPlugin({
                inject: "body",
                template: `${root} ${solutionName}/Views/Shared/_LayoutTemplate.cshtml`,
                filename: `${root}${solutionName}/${projectName}/Views/Shared/_Layout.cshtml`})`
